pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -123,9 +125,18 @@ func (admitter *VMICreateAdmitter) Admit(ar *admissionv1.AdmissionReview) *admis
 
 	reviewResponse := admissionv1.AdmissionResponse{}
 	reviewResponse.Allowed = true
+	reviewResponse.Warnings = append(reviewResponse.Warnings, warnDeprecatedAPIs(vmi)...)
 	return &reviewResponse
 }
 
+func warnDeprecatedAPIs(vmi *v1.VirtualMachineInstance) []string {
+	var warnings []string
+	if util.IsPasstVMI(vmi) {
```

Comment:
This does not seem to be a long lasting method to support deprecated features.
I suggest to prepare a mechanism that on one side registers features for deprecation and on the other side (here) acts to add the warning.

Whatever the solution is, I would expect to specify the deprecation once for a feature, without the need to repeat it in many places.

Reply:
I dont think we should invest in long lasting method for it at the moment, it should live until next release but yeah nothing prevents from the next dev to add more to it..

Reply:
I am not familiar with the step of removing the warnings in any near future.

While there are cases where we can start small, this one is cross SIG, covering many potential contributors. I would not like to see multiple methods to do the same, especially when we already know of a project wide need to deprecate features.

As I see it, the investment is not that big and this change as a whole is not that urgent to worth implementing it like this.